SpatialGDK/Source/SpatialGDK/Private/Interop/CrossServerRPCSender.cpp

```diff
@@ -24,8 +28,20 @@ void CrossServerRPCSender::SendCommand(const FUnrealObjectRef InTargetObjectRef,
 
 	CommandRequest CommandRequest(SpatialConstants::SERVER_TO_SERVER_COMMAND_ENDPOINT_COMPONENT_ID,
 								  SpatialConstants::UNREAL_RPC_ENDPOINT_COMMAND_ID);
-	RPCPayload::WriteToSchemaObject(CommandRequest.GetRequestObject(), InTargetObjectRef.Offset, Info.Index, FMath::RandHelper(INT_MAX),
+
+	uint64 UniqueRPCId = FMath::RandHelper(INT_MAX);
+	RPCPayload::WriteToSchemaObject(CommandRequest.GetRequestObject(), InTargetObjectRef.Offset, Info.Index, UniqueRPCId,
 									InPayload.PayloadData.GetData(), InPayload.PayloadData.Num());
+
+	FSpatialGDKSpanId SpanId;
+	if (EventTracer)
+	{
+		SpanId = EventTracer->TraceEvent(
+			FSpatialTraceEventBuilder::CreateSendCrossServerRPC(
+				TargetObject, Function, EventTraceUniqueId::GenerateForCrossServerRPC(InTargetObjectRef.Entity, UniqueRPCId)),
+			EventTracer->GetFromStack().GetConstId(), 1);
```

Comment:
Cool!

EventTracer->GetFromStack() allows us to latch onto user events right?

Reply:
Yep! They will be pushed here around the delegate call (it uses the same RPC tracking code for regular RPCs)

Reply:
Usually prefer adding `/* VarName */` to an undeclared var where possible, lil easier to maintain. Like here would be `/* LinearTraceId*/`